doc/conf.py

```diff
@@ -891,6 +894,56 @@ def setup(app):
     }
 
 
+def infer_next_release_versions():
+    """Infer the most likely next release versions to make."""
+    all_version_full = {"rc": "0.99.0rc1", "final": "0.99.0", "bf": "0.98.1"}
+    all_version_main = {"rc": "0.99", "final": "0.99", "bf": "0.98"}
+    all_previous_tag = {"rc": "0.98.micro", "final": "0.98.micro", "bf": "0.97.micro"}
+
+    try:
+        # Fetch the version switcher JSON; see `html_theme_options` for more details
+        versions_json = json.loads(
+            urlopen(html_theme_options["switcher"]["json_url"], timeout=10).read()
+        )
+        stable_version_index, stable_version = None, None
+        for i, ver in enumerate(versions_json):
+            if ver.get("preferred"):
+                stable_version_index = i
+                stable_version = parse(ver["version"])
+        assert stable_version_index is not None  # By version switcher spec
```

Comment:
In RC stage I think dev and stable are two minor versions apart? So there could be for instance 1.7.dev0 (dev), 1.6.0rc1, 1.5.x (stable)?

Reply:
I may need to double check how I implemented the version switcher logic 🤔

Reply:
Okay I think I find a snapshot somewhere and the logic is that dev and stable are always the first two. I will update this part of the logic :)

Reply:
You're right though that in `1.6.0rc1` RC stage, dev becomes 1.7.dev0 so the `Major/Minor final` tab will propose `1.7.0` whereas we want `1.6.0`.

Maybe we should just inspect the stable version, `v` in the following and
- for rc, it's `{v.major}.{v.minor + 1}.{0rc1}`
- for final, it's `{v.major}.{v.minor + 1}.{0}`
- for bf, it's `{v.major}.{v.minor}.{v.micro + 1}`

With that there's still a caveat. When we decide to make sklearn 2.0.0, it will propose 1.7.0 instead. But I think that's alright because we release major version very rarely, and the one time we dot it, it's not that problematic manually set the right version number in the commands.

Reply:
Gotcha let's keep the logic simple